assets/src/models/createDetourMachine.ts

```diff
@@ -297,30 +343,64 @@ export const createDetourMachine = setup({
                   },
                   {
                     actions: "detour.remove-last-waypoint",
+                    reenter: true,
                     target: "Place Waypoint",
                   },
                 ],
               },
             },
             "Finished Drawing": {
-              invoke: {
-                src: "fetch-finished-detour",
-                input: ({
-                  context: { routePattern, startPoint, endPoint },
-                }) => ({
-                  routePatternId: routePattern?.id,
-                  startPoint,
-                  endPoint,
-                }),
+              invoke: [
+                {
+                  src: "fetch-finished-detour",
+                  input: ({
+                    context: { routePattern, startPoint, endPoint },
+                  }) => ({
+                    routePatternId: routePattern?.id,
+                    startPoint,
+                    endPoint,
+                  }),
 
-                onDone: {
-                  actions: assign({
-                    finishedDetour: ({ event }) => event.output,
+                  onDone: {
+                    actions: assign({
+                      finishedDetour: ({ event }) => event.output,
+                    }),
+                  },
+
+                  onError: {},
+                },
+
+                {
+                  src: "fetch-detour-directions",
+                  input: ({
+                    context: { startPoint, waypoints, endPoint },
+                  }) => ({
+                    points: (startPoint ? [startPoint] : [])
+                      .concat(waypoints || [])
+                      .concat(endPoint ? [endPoint] : []),
                   }),
+
+                  onDone: {
+                    actions: assign({
+                      detourShape: ({ event }) => {
+                        const detourShape = event.output
+
+                        return map(detourShape, (shape) => {
+                          return {
+                            ...shape,
+                            directions: shape.directions?.concat({
+                              instruction: "Regular Route",
+                            }),
+                          }
+                        })
+                      },
```

Comment:
nit/suggestion(**non-blocking**): Just wondering, do you prefer this syntax over something like this?
```suggestion
                      detourShape: ({ event: { output: detourShape } }) =>
                        map(detourShape, (shape) => ({
                          ...shape,
                          directions: shape.directions?.concat({
                            instruction: "Regular Route",
                          }),
                        })),
```